analysis_request_cli/create_request.py

```diff
@@ -0,0 +1,124 @@
+# limitations under the License.
+
+import argparse
+import sys
+import os
+import requests
+from requests.auth import HTTPBasicAuth
+import json
+
+from mgnify_backlog import mgnify_handler
+from ena_portal_api import ena_handler
+from django.core.exceptions import ObjectDoesNotExist
+
+import logging
+
+logging.basicConfig(level=logging.INFO)
+
+API_URL = 'https://www.ebi.ac.uk/metagenomics/api/v1/'
+LOGIN_URL = 'https://www.ebi.ac.uk/metagenomics/api/http-auth/login/'
+LOGIN_FORM = 'https://www.ebi.ac.uk/metagenomics/api/http-auth/login_form'
+
+
+def parse_args(args):
+    parser = argparse.ArgumentParser(description='Tool to store analysis requests in the EMG backlog database')
+    parser.add_argument('study', help='Study accession (all are supported)')
+    parser.add_argument('webinID', help='Webin id')
+    parser.add_argument('RTticket', help='RT ticket')
+    parser.add_argument('--dev', action='store_true')
+    parser.add_argument('--private', action='store_true')
+    parser.add_argument('--priority', type=int, choices=range(0, 5), default=0)
+    parser.add_argument('--lineage', help='Full lineage of biome')
+    return parser.parse_args(args)
+
+
+# Get secondary accession of study from MGnify API
+def get_study(webin_id, mgys):
+    password = os.environ['MGNIFY_API_PASSWORD']
+    with requests.Session() as s:
+        get_csrftoken = s.get(LOGIN_FORM).headers
+        if 'csrftoken' in s.cookies:
+            # Django 1.6 and up
+            csrftoken = s.cookies['csrftoken']
+        else:
+            # older versions
+            csrftoken = s.cookies['csrf']
+
+        s.headers.update({'referer': 'https://www.ebi.ac.uk'})
+        login_data = {'username': webin_id, 'password': password, 'csrfmiddlewaretoken': csrftoken}
+        login = s.post(LOGIN_URL, data=login_data)
+
+        req = s.get(os.path.join(API_URL, 'studies', mgys), auth=HTTPBasicAuth(webin_id, password))
+        study = json.loads(req.text)
+        try:
+            return study['data']['attributes']['secondary-accession']
+        except KeyError as e:
+            logging.error(study)
+            raise EnvironmentError('API response to study query was not valid (try again)')
+
+
+# Remove runs from set if they were already analysed with latest pipeline
+def filter_duplicate_runs(annotated_runs, study_runs):
+    annotated_run_accession = [run.primary_accession for run in annotated_runs]
+    return list(filter(lambda r: r['run_accession'] not in annotated_run_accession, study_runs))
+
+
+def main(argv=None):
+    args = parse_args(argv)
+    mh = mgnify_handler.MgnifyHandler('dev' if args.dev else 'prod')
+    ena = ena_handler.EnaApiHandler()
+    try:
+        user = mh.get_user(args.webinID)
+    except ObjectDoesNotExist:
+        logging.warning('User {} not in db, creating user.'.format(args.webinID))
+        user = mh.create_user(webin_id=args.webinID, email='unknown@unknown.com', first_name='first', surname='last',
```

Comment:
We need to pull the email and the user name as well. Not sure if that is possible using the API. If not we need to create a ENA database handler.